### PR TITLE
Added tests for #15814

### PR DESCRIPTION
**Pull request by leandrogehlen**
Added tests for truncate table via fixture

### CODE REVIEW

**Comment by leandrogehlen** [comment]
@samdark, the `ArrayFixtureTest::testWrongDataFileException` is a test to validate this exception.

It seems a little strange that's `ActiveFixture` to have a different behavior of `ArrayFixture`.

Do you want remove this exception for  ArrayFixture too or remove this excption only for ActiveFixture ?

**Comment by samdark** [comment]
For now I want it to be compatible to pre-2.0.14.

**Review by samdark — APPROVED** [review]
Please add a line for changelog.

File: framework/test/FileFixtureTrait.php [hunk 1]
```diff
@@ -35,25 +35,26 @@
...
+     * @param bool $throwException whether to throw exception if fixture data file does not exist.
      * @return array the data to be put into the database
      * @throws InvalidConfigException if the specified data file does not exist.
      */
-    protected function loadData($file)
+    protected function loadData($file, $throwException = true)
     {
-        if ($file === false || $file === null) {
-            return [];
+        if ($file) {
```
Comment (alexraputa) [inline framework/test/FileFixtureTrait.php]: @leandrogehlen, @samdark How about this structure of the method?

```php
if (!$file) {
    return [];
}

if (basename($file) === $file && $this->dataDirectory !== null) {
    $file = $this->dataDirectory . '/' . $file;
}

$file = Yii::getAlias($file);
if (is_file($file)) {
    return require $file;
} elseif ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
}
```

Comment (rugabarbo) [reply framework/test/FileFixtureTrait.php]:
```php
if (!$file) {
    return [];
}

if (basename($file) === $file && $this->dataDirectory !== null) {
    $file = $this->dataDirectory . '/' . $file;
}

$file = Yii::getAlias($file);
if (is_file($file)) {
    return require $file;
}

if ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
}
```

Comment (leandrogehlen) [reply framework/test/FileFixtureTrait.php]: I think this not works, when file does not exists

Comment (leandrogehlen) [reply framework/test/FileFixtureTrait.php]: must be :

```php
if (!$file) {
    return [];
}

if (basename($file) === $file && $this->dataDirectory !== null) {
    $file = $this->dataDirectory . '/' . $file;
}

$file = Yii::getAlias($file);
if (is_file($file)) {
    return require $file;
} elseif ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
} else {
   return [];
}

Comment (alexraputa) [reply framework/test/FileFixtureTrait.php]: I think these `else` statement and `return []` are unnecessary. The test should be green after the implementation of my example.

```php
if (is_file($file)) {
    return require $file;
} elseif ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
} else {
   return [];
}
```

**Comment by leandrogehlen** [comment]
@samdark, maybe something like:

```php
$result = [];

if (basename($file) === $file && $this->dataDirectory !== null) {
    $file = $this->dataDirectory . '/' . $file;
}

$file = Yii::getAlias($file);
if (is_file($file)) {
    $result = require $file;
} elseif ($throwException) {
    throw new InvalidConfigException("Fixture data file does not exist: {$file}");
}
return $result;

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**